CMakeLists.txt

```diff
@@ -2,7 +2,21 @@ cmake_minimum_required(VERSION 2.8.12)
 project(MINEDMAP CXX)
 
 
-find_package(PNG REQUIRED)
+# search for pkg-config
+include (FindPkgConfig)
+if (NOT PKG_CONFIG_FOUND)
+    message (FATAL_ERROR "pkg-config not found")
+endif ()
+
+# check for libpng
+pkg_check_modules (LIBPNG libpng16 REQUIRED)
+if (NOT LIBPNG_FOUND)
+    message(FATAL_ERROR "You don't seem to have libpng16 development libraries installed")
+else ()
+    include_directories (${LIBPNG_INCLUDE_DIRS})
+    link_directories (${LIBPNG_LIBRARY_DIRS})
+    link_libraries (${LIBPNG_LIBRARIES})
+endif ()
```

Comment:
I don't see an issue with switching to pkg-config, but your code has a few issues.

I have pushed an alternative patch that resolves zlib though pkg-config as well (so we don't end up with two different versions of zlib). In addition, the StackOverflow answer you found is from 2015, and CMake has developed considerably in the mean time. Please check if the current `main` branch finds the libraries correctly on OSX now.

Reply:
What about switching to bazel and then the correct libpng dependencies would get pulled automatically from e.g. github?

Reply:
These days my build system of choice for C/C++ is Meson, which can optionally download dependencies as well. For MinedMap, I'm not sure if switching the build system makes sense, as I intend to rewrite it in Rust anyways at some point...